### PR TITLE
BAU Java Spotless check

### PR DESCRIPTION
**Pull request by idrop**
## Proposed changes

A java linter that runs before the `gradle compileJava` task, so really up in your grille. It'll also get run when a PR is created or updated.

This could pair well with a pre-push and/or pre-commit git hook.

Fixes can be applied with `./gradlew :spotlessApply` and are the 2nd commit on this PR.

More info on spotless here: https://github.com/diffplug/spotless/tree/main/plugin-gradle#google-java-format

### CODE REVIEW

**Review by Wynndow — DISMISSED** [review]
I'm in for this. Although I wouldn't want it on a pre-commit or push hook. Don't want to get blocked from doing things quickly. Part of the PR checks would be good though.

File: build.gradle [hunk 1]
```diff
@@ -57,3 +61,14 @@ sonarqube {
...
+
+spotless {
+    java {
+        target "**/*.java"
+        googleJavaFormat("1.12.0").aosp()
+        importOrder "", "javax", "java", "\\#"
+    }
+}
+
+compileJava.dependsOn "spotlessCheck"
```
Comment (jamietanna) [inline build.gradle]: This shouldn't be required, as it's bound into the `check` phase

File: build.gradle [hunk 2]
```diff
@@ -57,3 +61,14 @@ sonarqube {
         property "sonar.host.url", "https://sonarcloud.io"
     }
 }
+
+spotless {
+    java {
+        target "**/*.java"
```
Comment (jamietanna) [inline build.gradle]: Spotless team recommends this as it includes less code unnecessarily:

```suggestion
        target "src/**/*.java"
```

File: build.gradle [hunk 3]
```diff
@@ -1,8 +1,12 @@
+import org.gradle.api.JavaVersion
+import org.gradle.api.tasks.bundling.Zip
+
 plugins {
     id "java"
     id "application"
     id "idea"
     id "org.sonarqube" version "3.3"
+    id "com.diffplug.spotless" version "5.17.1"
```
Comment (jamietanna) [inline build.gradle]: Can we not bump to a later version?

File: build.gradle [hunk 4]
```diff
@@ -1,8 +1,12 @@
+import org.gradle.api.JavaVersion
+import org.gradle.api.tasks.bundling.Zip
```
Comment (jamietanna) [inline build.gradle]: were these needed?

File: build.gradle [hunk 5]
```diff
@@ -57,3 +61,14 @@ sonarqube {
         property "sonar.host.url", "https://sonarcloud.io"
     }
 }
+
+spotless {
```
Comment (jamietanna) [inline build.gradle:71]: should we also add `groovyGradle` formatting?

**Review by kerrr — APPROVED** [review]

**Pull request closed without merging**